src/commp.rs

```diff
@@ -0,0 +1,261 @@
+}
+
+fn padded<R: Sized>(inp: &mut R, size: u64) -> PadReader<&mut R>
+where
+    R: io::Read,
+{
+    let padded_size = padded_size(size);
+
+    let pad_reader = PadReader {
+        size: usize::try_from(size).unwrap(),
+        padsize: usize::try_from(padded_size).unwrap(),
+        pos: 0,
+        inp: inp,
+    };
+
+    return pad_reader;
+}
+
+#[allow(dead_code)]
+pub fn generate_commp_filecoin_proofs<R: Sized>(
+    inp: &mut R,
+    size: u64,
+) -> Result<CommP, std::io::Error>
+where
+    R: io::Read,
+{
+    let pad_reader = padded(inp, size);
+    let padded_size = pad_reader.padsize;
+
+    let info = generate_piece_commitment(pad_reader, UnpaddedBytesAmount(padded_size as u64))
+        .expect("failed to generate piece commitment");
+
+    Ok(CommP {
+        padded_size: padded_size as u64,
+        piece_size: u64::from(info.size),
+        bytes: info.commitment,
+    })
+}
+
+#[allow(dead_code)]
+pub fn generate_commp_storage_proofs<R: Sized>(
+    inp: &mut R,
+    size: u64,
+) -> Result<CommP, std::io::Error>
+where
+    R: io::Read,
+{
+    let pad_reader = padded(inp, size);
+    let padded_size = pad_reader.padsize;
+
+    // Grow the vector big enough so that it doesn't grow it automatically
+    let mut data = Vec::with_capacity((padded_size as u64 as f64 * 1.01) as usize);
+    let mut temp_piece_file = Cursor::new(&mut data);
+
+    // send the source through the preprocessor, writing output to temp file
+    let piece_size =
+        UnpaddedBytesAmount(write_padded(pad_reader, &mut temp_piece_file).unwrap() as u64);
+    temp_piece_file.seek(SeekFrom::Start(0))?;
+    let commitment = generate_piece_commitment_bytes_from_source::<DefaultPieceHasher>(
+        &mut temp_piece_file,
+        PaddedBytesAmount::from(piece_size).into(),
+    );
+
+    Ok(CommP {
+        padded_size: padded_size as u64,
+        piece_size: u64::from(piece_size),
+        bytes: commitment.unwrap(),
+    })
+}
+
+pub fn generate_commp_storage_proofs_mem<R: Sized>(
+    inp: &mut R,
+    size: u64,
+    multistore: bool,
+) -> Result<CommP, std::io::Error>
+where
+    R: io::Read,
+{
+    let pad_reader = padded(inp, size);
+    let padded_size = pad_reader.padsize;
+
+    // Grow the vector big enough so that it doesn't grow it automatically
+    let fr32_capacity = (padded_size as f64) * 1.008; // rounded up extra space for 2 in every 254 bits
+    info!(
+        "Padded size = {}, allocating vector with fr32 size = {}",
+        padded_size, fr32_capacity
+    );
+    let mut data = Vec::with_capacity(fr32_capacity as usize);
+    let mut temp_piece_file = Cursor::new(&mut data);
+
+    // send the source through the preprocessor, writing output to temp file
+    let piece_size =
+        UnpaddedBytesAmount(write_padded(pad_reader, &mut temp_piece_file).unwrap() as u64);
+    info!("Piece size = {:?}", piece_size);
+    temp_piece_file.seek(SeekFrom::Start(0))?;
+    let commitment: Fr32Ary;
+
+    if multistore {
```

Comment:
You can use implicit returns in if statements. So this would be:

```rust
let commitment = if multistore {
    local_multistore_generate_piece_commitment_bytes_from_source::<
            DefaultPieceHasher,
            Cursor<&mut Vec<u8>>,
        >(
            &mut temp_piece_file,
            PaddedBytesAmount::from(piece_size).into(),
        )?
} else {
…
};
```